src/systems/scene_broadcaster/SceneBroadcaster.cc

```diff
@@ -294,6 +303,18 @@ void SceneBroadcaster::PostUpdate(const UpdateInfo &_info,
       this->dataPtr->stateCv.notify_all();
     }
 
+    // process async state requests
+    if (!this->dataPtr->stateRequests.empty())
+    {
+      ignition::msgs::SerializedStepMap res;
```

Comment:
Do we need to do this extra copy? It seems that `stateMutex` protects it from being accessed while we send the requests. Ignition Transport makes its own copy when you call `Request()` and the service provider is on a separate process (it doesn't copy if the service provider is on the same process.

Reply:
I see, we won't need to copy then. Removed copy in ab948fb